src/utils/pgwire/src/pg_server.rs

```diff
@@ -30,6 +30,8 @@ pub trait SessionManager: Send + Sync + 'static {
     type Session: Session;
 
     fn connect(&self, database: &str) -> Result<Arc<Self::Session>, BoxedError>;
+
+    fn check_db_name(&self, database: &str) -> bool;
```

Comment:
It's unnecessary to be defined in trait, you should impl it simply in SessionManagerImpl.

```
impl SessionManagerImpl {
  fn connect(&self, database: &str) {
    self.check_db_name()
  }
```